bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -140,6 +143,102 @@ public static PercentType[] hsbToRgbPercent(HSBType hsb) {
         return new PercentType[] { red, green, blue };
     }
 
+    /**
+     * Transform <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType} to RGBW.
+     *
+     * See <a href=
+     * "https://stackoverflow.com/questions/40312216/converting-rgb-to-rgbw">Converting RGB to RGBW</a>.
+     *
+     * This function does not round the components. For conversion to integer values in the range 0 to 255 use
+     * {@link #hsbToRgb(HSBType)}.
+     *
+     * See also: {@link #hsbToRgb(HSBType)}, {@link #hsbTosRgb(HSBType)}, {@link #hsbToRgbPercent(HSBType)}
+     *
+     * @param hsb an {@link HSBType} value.
+     * @return array of four {@link PercentType} with the RGBW values in the range 0 to 100 percent.
+     */
+    public static PercentType[] hsbToRgbwPercent(HSBType hsb) {
+        int[] rgb = hsbToRgb(hsb);
```

Comment:
If you convert to `BigDecimal` afterwards (and do other calculations), why don't you use `hsbToRgbPercent`? You can use `.toBigDecimal()` which avoids creation of a new `BigDecimal`.

Reply:
I need the value in byte from 0 to 255. When i convert `PercentType.toBigDecimal()` i have only 0-100.

Reply:
I will check the calculation think, is a good idea.